### ISSUE TITLE
copy_async no longer exists - playlist file export breaks

### ISSUE THREAD

**Issue by virtuald** [exaile/exaile#198]

**Comment by javiercantero** [comment]
The error message is:
```
Traceback (most recent call last):
  File ".../exaile/xlgui/widgets/dialogs.py", line 1616, in <lambda>
    dialog.connect( 'uris-selected', lambda widget, uris: _on_uri(uris[0]))
  File ".../exaile/xlgui/widgets/dialogs.py", line 1611, in _on_uri
    dialog.do_copy()
  File ".../exaile/xlgui/widgets/dialogs.py", line 1394, in do_copy
    self._start_next_copy()
  File ".../exaile/xlgui/widgets/dialogs.py", line 1462, in _start_next_copy
    self.source.copy_async( self.destination, self._finish_single_copy, flags=flags, cancellable=self.cancel ) 
AttributeError: 'GLocalFile' object has no attribute 'copy_async'
```
The problem is that the Gio function `g_file_copy_async()` is not introspectable in the current gobject-introspection package, hence it can be accessed using python gi. That's the situation since the commit GNOME/glib@89b558077f97c630c5b26b3527a6c7d0515f1f2c in which they disable it (the "(skip)" bit in the annotated function). Note that commit is from 2010.

The cause of such omission is harder to trace, but I think the GNOME bug [616044](https://bugzilla.gnome.org/show_bug.cgi?id=616044) explains it, plus its duplicate [708297](https://bugzilla.gnome.org/show_bug.cgi?id=708297) and the comments in [746111](https://bugzilla.gnome.org/show_bug.cgi?id=746111). In summary, they don't know how to handle functions with two callbacks and two different scopes for the closure variables. The Ubuntu bug [1033196](https://bugs.launchpad.net/ubuntu/+source/gobject-introspection/+bug/1033196) (from 2012) supports this too.

I see three possible solutions:

- Push for upstream to solve their bug and make `copy_async` available to everybody using g-i. That's the best-case scenario, but unfortunately I doubt this is going to happen soon (if happened) since 7 years have already been passed and nobody has done anything yet.
- Use the python machinery to write a non-blocking copy. For example, do the copy in a python thread should suffice. The problem is that adding more threads to the current situation (meaning #83) might not be such a good idea.
- Since 'Export Files' is not a essential feature, disable it for Exaile 4.0 and delay any solution to 4.0.1 or a later version.

**Comment by virtuald** [comment]
I was hoping there would be a backdoor way to get copy_async to work (or access it via some other call perhaps), but perhaps that is just a pipe dream.

Lacking that, I think I would be in favor of solution 2, using something like [ProgressThread](https://github.com/exaile/exaile/blob/master/xl/common.py#L684). Adding more threads isn't bad as long as they don't do things they aren't supposed to do.

**Comment by virtuald** [comment]
Closing this as it was fixed in #316
